dali/test/python/test_operator_power_spectrum.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from nvidia.dali.pipeline import Pipeline
+import nvidia.dali.ops as ops
+import nvidia.dali.types as types
+import nvidia.dali as dali
+import numpy as np
+from numpy.testing import assert_array_equal, assert_allclose
+from functools import partial
+from test_utils import check_batch
+from test_utils import compare_pipelines
+from test_utils import RandomDataIterator
+
+class PowerSpectrumPipeline(Pipeline):
+    def __init__(self, device, batch_size, iterator, axis, nfft, num_threads=1, device_id=0):
+        super(PowerSpectrumPipeline, self).__init__(batch_size, num_threads, device_id)
+        self.device = device
+        self.iterator = iterator
+        self.inputs = ops.ExternalSource()
+        self.fft = ops.PowerSpectrum(device=self.device, axis=axis, nfft=nfft)
+
+    def define_graph(self):
+        self.data = self.inputs()
+        out = self.data.gpu() if self.device == 'gpu' else self.data
+        out = self.fft(out)
+        return out
+
+    def iter_setup(self):
+        data = self.iterator.next()
+        self.feed_input(self.data, data)
+
+def power_spectrum_numpy(nfft, axis, waveform):
+    fft_out = np.fft.fft(waveform, axis=axis, n=nfft)
+    power_spectrum = fft_out.real ** 2 + fft_out.imag ** 2
+    shape = waveform.shape
+
+    out_shape = list(shape)
+    out_shape[axis] = nfft/2+1
```

Comment:
It will different between python2 and python3

Reply:
can you elaborate?